zipline/pipeline/expression.py

```diff
@@ -248,7 +248,12 @@ def _rebind_variables(self, new_inputs):
         new_inputs.
         """
         expr = self._expr
-        for idx, input_ in enumerate(self.inputs):
+
+        # This call to `reversed` is required to handle the case in which the
+        # length of self.inputs is greater than 10. Otherwise, expressions
+        # containing an x_10 will be improperly overwritten at the same time
+        # that x_1 is overwritten.
```

Comment:
The essence of the bug here is that some of our variable names are substrings of other variable names, so when we replace `x_1` before `x_10`, we inadvertently modify `x_10` because it contains `x_1` as a substring.  By enumerating in reverse order, we ensure that every variable name which is a substring of another variable name is processed **after** the variable of which it is a substring.  This guarantees that the substitution of any given variable index only every affects exactly its own index.